### Algoritimos de classificação

base de dados: https://www.kaggle.com/harshitshankhdhar/imdb-dataset-of-top-1000-movies-and-tv-shows

Data de início: 28/03/2021 <br>
Autor: Douglas  <br>

Objetivo: Realizar analise exploratoria dos dados da base de filmes e tentar criar um modelo de classificação;
<br>

Vamos testar duas abordagem: predição da categoria do filme baseado nas variaveis independentes e depois baseado unicamente na descrião

Content
Data:-

Poster_Link - Link of the poster that imdb using <br>
Series_Title = Name of the movie <br>
Released_Year - Year at which that movie released <br>
Certificate - Certificate earned by that movie <br>
Runtime - Total runtime of the movie <br>
Genre - Genre of the movie <br>
IMDB_Rating - Rating of the movie at IMDB site <br>
Overview - mini story/ summary <br>
Meta_score - Score earned by the movie <br>
Director - Name of the Director <br>
Star1,Star2,Star3,Star4 - Name of the Stars <br>
Noofvotes - Total number of votes <br>
Gross - Money earned by that movie <br>


In [1]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

In [2]:
path = '../Projects/Datasets/imdb_top_1000.csv'

In [3]:
df = pd.read_csv('../Projects/Datasets/imdb_top_1000.csv')

In [4]:
df.head()

Poster_Link  \
0  https://m.media-amazon.com/images/M/MV5BMDFkYT...   
1  https://m.media-amazon.com/images/M/MV5BM2MyNj...   
2  https://m.media-amazon.com/images/M/MV5BMTMxNT...   
3  https://m.media-amazon.com/images/M/MV5BMWMwMG...   
4  https://m.media-amazon.com/images/M/MV5BMWU4N2...   

               Series_Title Released_Year Certificate  Runtime  \
0  The Shawshank Redemption          1994           A  142 min   
1             The Godfather          1972           A  175 min   
2           The Dark Knight          2008          UA  152 min   
3    The Godfather: Part II          1974           A  202 min   
4              12 Angry Men          1957           U   96 min   

                  Genre  IMDB_Rating  \
0                 Drama          9.3   
1          Crime, Drama          9.2   
2  Action, Crime, Drama          9.0   
3          Crime, Drama          9.0   
4          Crime, Drama          9.0   

                                            Overview  Meta_score  \
0  Two imprisoned men bond over a number of years...        80.0   
1  An organized crime dynasty's aging patriarch t...       100.0   
2  When the menace known as the Joker wreaks havo...        84.0   
3  The early life and career of Vito Corleone in ...        90.0   
4  A jury holdout attempts to prevent a miscarria...        96.0   

               Director           Star1           Star2          Star3  \
0        Frank Darabont     Tim Robbins  Morgan Freeman     Bob Gunton   
1  Francis Ford Coppola   Marlon Brando       Al Pacino     James Caan   
2     Christopher Nolan  Christian Bale    Heath Ledger  Aaron Eckhart   
3  Francis Ford Coppola       Al Pacino  Robert De Niro  Robert Duvall   
4          Sidney Lumet     Henry Fonda     Lee J. Cobb  Martin Balsam   

            Star4  No_of_Votes        Gross  
0  William Sadler      2343110   28,341,469  
1    Diane Keaton      1620367  134,966,411  
2   Michael Caine      2303232  534,858,444  
3    Diane Keaton      1129952   57,300,000  
4    John Fiedler       689845    4,360,000

In [5]:
df

Poster_Link  \
0    https://m.media-amazon.com/images/M/MV5BMDFkYT...   
1    https://m.media-amazon.com/images/M/MV5BM2MyNj...   
2    https://m.media-amazon.com/images/M/MV5BMTMxNT...   
3    https://m.media-amazon.com/images/M/MV5BMWMwMG...   
4    https://m.media-amazon.com/images/M/MV5BMWU4N2...   
5    https://m.media-amazon.com/images/M/MV5BNzA5ZD...   
6    https://m.media-amazon.com/images/M/MV5BNGNhMD...   
7    https://m.media-amazon.com/images/M/MV5BNDE4OT...   
8    https://m.media-amazon.com/images/M/MV5BMjAxMz...   
9    https://m.media-amazon.com/images/M/MV5BMmEzNT...   
10   https://m.media-amazon.com/images/M/MV5BN2EyZj...   
11   https://m.media-amazon.com/images/M/MV5BNWIwOD...   
12   https://m.media-amazon.com/images/M/MV5BOTQ5ND...   
13   https://m.media-amazon.com/images/M/MV5BZGMxZT...   
14   https://m.media-amazon.com/images/M/MV5BNzQzOT...   
15   https://m.media-amazon.com/images/M/MV5BY2NkZj...   
16   https://m.media-amazon.com/images/M/MV5BYmU1ND...   
17   https://m.media-amazon.com/images/M/MV5BZjA0OW...   
18   https://m.media-amazon.com/images/M/MV5BNjViNW...   
19   https://m.media-amazon.com/images/M/MV5BYWZjMj...   
20   https://m.media-amazon.com/images/M/MV5BOTc2ZT...   
21   https://m.media-amazon.com/images/M/MV5BZjdkOT...   
22   https://m.media-amazon.com/images/M/MV5BOTMwYj...   
23   https://m.media-amazon.com/images/M/MV5BMjlmZm...   
24   https://m.media-amazon.com/images/M/MV5BZjhkMD...   
25   https://m.media-amazon.com/images/M/MV5BMTUxMz...   
26   https://m.media-amazon.com/images/M/MV5BYmJmM2...   
27   https://m.media-amazon.com/images/M/MV5BOTUwOD...   
28   https://m.media-amazon.com/images/M/MV5BNjNhZT...   
29   https://m.media-amazon.com/images/M/MV5BNzVlY2...   
..                                                 ...   
970  https://m.media-amazon.com/images/M/MV5BMTQxND...   
971  https://m.media-amazon.com/images/M/MV5BMTY5Nj...   
972  https://m.media-amazon.com/images/M/MV5BNjg5ZD...   
973  https://m.media-amazon.com/images/M/MV5BMzFkM2...   
974  https://m.media-amazon.com/images/M/MV5BNWFlYW...   
975  https://m.media-amazon.com/images/M/MV5BMjE0OD...   
976  https://m.media-amazon.com/images/M/MV5BN2JlZT...   
977  https://m.media-amazon.com/images/M/MV5BODk1ZW...   
978  https://m.media-amazon.com/images/M/MV5BM2I1ZW...   
979  https://m.media-amazon.com/images/M/MV5BZTllNW...   
980  https://m.media-amazon.com/images/M/MV5BZmI5Yz...   
981  https://m.media-amazon.com/images/M/MV5BNWQ4MG...   
982  https://m.media-amazon.com/images/M/MV5BN2VlNj...   
983  https://m.media-amazon.com/images/M/MV5BYTU2MW...   
984  https://m.media-amazon.com/images/M/MV5BMGQ0OG...   
985  https://m.media-amazon.com/images/M/MV5BNDQ3Mz...   
986  https://m.media-amazon.com/images/M/MV5BMzZiOD...   
987  https://m.media-amazon.com/images/M/MV5BNDU1Mj...   
988  https://m.media-amazon.com/images/M/MV5BMjM1Nj...   
989  https://m.media-amazon.com/images/M/MV5BYzZhOD...   
990  https://m.media-amazon.com/images/M/MV5BYjRmY2...   
991  https://m.media-amazon.com/images/M/MV5BMzAyND...   
992  https://m.media-amazon.com/images/M/MV5BMjAwMT...   
993  https://m.media-amazon.com/images/M/MV5BYTE4YW...   
994  https://m.media-amazon.com/images/M/MV5BZjQyMG...   
995  https://m.media-amazon.com/images/M/MV5BNGEwMT...   
996  https://m.media-amazon.com/images/M/MV5BODk3Yj...   
997  https://m.media-amazon.com/images/M/MV5BM2U3Yz...   
998  https://m.media-amazon.com/images/M/MV5BZTBmMj...   
999  https://m.media-amazon.com/images/M/MV5BMTY5OD...   

                                          Series_Title Released_Year  \
0                             The Shawshank Redemption          1994   
1                                        The Godfather          1972   
2                                      The Dark Knight          2008   
3                               The Godfather: Part II          1974   
4                                         12 Angry Men          1957   
5        The L

### Ajuste da variavel resposta - <br>
Lets start do know our data. First we're going to exam the Genere and think about our respose variable.

In [6]:
df.groupby('Genre').mean()

IMDB_Rating  Meta_score    No_of_Votes
Genre                                                               
Action, Adventure                8.180000   71.800000  925533.400000
Action, Adventure, Biography     7.900000         NaN   52397.000000
Action, Adventure, Comedy        7.910000   66.857143  456076.600000
Action, Adventure, Crime         7.600000         NaN   63882.000000
Action, Adventure, Drama         8.150000   80.461538  663989.928571
Action, Adventure, Family        7.600000   78.000000  474827.000000
Action, Adventure, Fantasy       8.200000   78.333333  954767.666667
Action, Adventure, History       7.900000   85.000000  173999.000000
Action, Adventure, Horror        7.900000   71.000000  111512.000000
Action, Adventure, Mystery       7.600000   57.000000  583158.000000
Action, Adventure, Romance       7.900000   97.000000   47175.000000
Action, Adventure, Sci-Fi        7.928571   71.523810  696942.761905
Action, Adventure, Thriller      7.740000   78.400000  408529.800000
Action, Adventure, War           7.700000   68.000000   59548.000000
Action, Adventure, Western       7.800000   85.500000   82560.000000
Action, Biography, Crime         7.866667   81.000000   85794.333333
Action, Biography, Drama         7.942857   67.666667  192873.000000
Action, Comedy, Crime            8.016667   68.000000  244859.166667
Action, Comedy, Fantasy          7.766667   73.666667  210340.666667
Action, Comedy, Mystery          7.800000   81.000000  463466.000000
Action, Comedy, Romance          8.200000         NaN   45642.500000
Action, Crime, Comedy            8.000000         NaN   27721.000000
Action, Crime, Drama             7.880000   73.434783  284026.566667
Action, Crime, Mystery           7.600000   80.000000  508417.000000
Action, Crime, Thriller          7.800000   69.125000  168416.800000
Action, Drama                    7.940000   53.500000  269585.000000
Action, Drama, History           7.766667   77.000000  240970.000000
Action, Drama, Mystery           8.080000   72.000000  373034.600000
Action, Drama, Sci-Fi            8.150000   69.500000  840316.500000
Action, Drama, Sport             8.200000   71.000000  435950.000000
...                                   ...         ...            ...
Drama, Romance                   7.935135   82.687500  244279.891892
Drama, Romance, Sci-Fi           8.150000   89.500000  726218.000000
Drama, Romance, Thriller         7.850000   78.000000  159971.500000
Drama, Romance, War              8.025000   86.750000  173157.250000
Drama, Sci-Fi                    7.933333   88.250000  219994.333333
Drama, Sci-Fi, Thriller          7.700000   67.000000  493902.500000
Drama, Sport                     7.900000   74.666667  299882.833333
Drama, Thriller                  7.857143   75.727273  184477.428571
Drama, Thriller, War             8.100000   74.500000  185847.666667
Drama, Thriller, Western         7.900000   89.000000   97222.000000
Drama, War                       8.073333   82.923077  224782.466667
Drama, War, Western              7.600000   77.000000   30144.000000
Drama, Western                   7.980000   84.800000  429668.200000
Family, Fantasy, Musical         7.800000   67.000000  178731.000000
Family, Sci-Fi                   7.800000   91.000000  372490.000000
Fantasy, Horror                  7.900000         NaN   88794.000000
Fantasy, Horror, Mystery         8.100000         NaN   57428.000000
Film-Noir, Mystery               8.000000   96.000000  148928.000000
Film-Noir, Mystery, Thriller     8.100000   97.000000  158731.000000
Film-Noir, Thriller              7.800000   94.000000   59556.000000
Horror                           7.900000   84.500000  194700.000000
Horror, Mystery, Sci-Fi          8.100000   57.000000  371271.000000
Horror, Mystery, Thriller        7.850000   75.500000  453433.250000
Horror, Sci-Fi                   8.050000   88.000000  409244.500000
Horror, Thriller                 7.800000   88.000000  174831.500000
Mystery, Romance, Thriller  

How we see, there are movies with more than 1 Genere, so, lets create 3 generes columns

In [7]:
df[['Genre_1','Genre_2', 'Genre_3']] = df['Genre'].str.split(',',expand=True)


In [8]:
df['Genre_1'] = df['Genre_1'].str.replace(' ', '')
df['Genre_2'] = df['Genre_2'].str.replace(' ', '')
df['Genre_3'] = df['Genre_3'].str.replace(' ', '')

We have a problem here. Once that the genre is sort by alphabetic order some genre arent representative in the colum genre_1. Ex Thriller.We have 99 thriller classification in genre_3, but 1 classification in genre_1.

So my first abordage is duplicate the films with one or more genre. But lets see how this comporte:

In [9]:
total = df[['Genre_1', 'Genre_2', 'Genre_3']].isnull().sum()
print('Tamanho da nossa base:', len(df))
total

Tamanho da nossa base: 1000


Genre_1      0
Genre_2    105
Genre_3    354
dtype: int64

Grande parte da nossa base possui mais que 1 genero - (apenas 105 filmes nao possuem 2 generos e 354 nao possuem 3).

In [10]:
df_gen2 = df[df['Genre_2'].notna()]
df_gen3 = df[df['Genre_3'].notna()]
df_gen2['Genre_all'] = df_gen2['Genre_2']
df_gen3['Genre_all'] = df_gen2['Genre_3']
df['Genre_all'] = df['Genre_1']

C:\Users\dougg\Anaconda3\lib\site-packages\ipykernel_launcher.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  This is separate from the ipykernel package so we can avoid doing imports until
C:\Users\dougg\Anaconda3\lib\site-packages\ipykernel_launcher.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  after removing the cwd from sys.path.


In [11]:
df_all_movies = pd.concat([df, df_gen2, df_gen3])

In [12]:
len(df_all_movies)

2541

In [13]:
df_all_movies.groupby('Genre_all').count()

Poster_Link  Series_Title  Released_Year  Certificate  Runtime  \
Genre_all                                                                   
Action             189           189            189          175      189   
Adventure          196           196            196          189      196   
Animation           82            82             82           80       82   
Biography          109           109            109          107      109   
Comedy             233           233            233          212      233   
Crime              209           209            209          184      209   
Drama              724           724            724          644      724   
Family              56            56             56           53       56   
Fantasy             66            66             66           61       66   
Film-Noir           19            19             19           15       19   
History             56            56             56           53       56   
Horror              32            32             32           23       32   
Music               35            35             35           33       35   
Musical             17            17             17           16       17   
Mystery             99            99             99           87       99   
Romance            125           125            125          114      125   
Sci-Fi              67            67             67           64       67   
Sport               19            19             19           19       19   
Thriller           137           137            137          117      137   
War                 51            51             51           41       51   
Western             20            20             20           19       20   

           Genre  IMDB_Rating  Overview  Meta_score  Director  Star1  Star2  \
Genre_all                                                                     
Action       189          189       189         156       189    189    189   
Adventure    196          196       196         180       196    196    196   
Animation     82           82        82          75        82     82     82   
Biography    109          109       109          97       109    109    109   
Comedy       233          233       233         188       233    233    233   
Crime        209          209       209         169       209    209    209   
Drama        724          724       724         600       724    724    724   
Family        56           56        56          50        56     56     56   
Fantasy       66           66        66          58        66     66     66   
Film-Noir     19           19        19          11        19     19     19   
History       56           56        56          48        56     56     56   
Horror        32           32        32          27        32     32     32   
Music         35           35        35          29        35     35     35   
Musical       17           17        17          12        17     17     17   
Mystery       99           99        99          80        99     99     99   
Romance      125          125       125         104       125    125    125   
Sci-Fi        67           67        67          62        67     67     67   
Sport         19           19        19          17        19     19     19   
Thriller     137          137       137         118       137    137    137   
War           51           51        51          45        51     51     51   
Western       20           20        20          20        20     20     20   

           Star3  Star4  No_of_Votes  Gross  Genre_1  Genre_2  Genre_3  
Genre_all                                                               
Action       189    189          189    156      189      189      173  
Adventure    196    196          196    170      196      196      182  
Animation     82     82           82     67       82       82       81  
Biography    109    109          109     96      109    

### Variable Star -> Dummy

In [14]:
#Tratando variaveis:
#criar um arry com todas as estrelas:

df_all_movies['all_stars'] = df_all_movies['Star1'].str.replace(' ', '') + df_all_movies['Star2'].str.replace(' ', '') + df_all_movies['Star3'].str.replace(' ', '') + df_all_movies['Star4'].str.replace(' ', '') 

In [15]:
df_all_movies['all_stars'].head()

0       TimRobbinsMorganFreemanBobGuntonWilliamSadler
1            MarlonBrandoAlPacinoJamesCaanDianeKeaton
2    ChristianBaleHeathLedgerAaronEckhartMichaelCaine
3         AlPacinoRobertDeNiroRobertDuvallDianeKeaton
4          HenryFondaLeeJ.CobbMartinBalsamJohnFiedler
Name: all_stars, dtype: object

In [16]:
df_all_movies['all_stars2'] = df_all_movies[['Star1', 'Star2','Star3','Star4']].agg(','.join, axis=1)

In [17]:
df_all_movies['all_stars2'] = df_all_movies['all_stars2'].str.replace(' ', '')

In [18]:
df_all_movies['all_stars2'].head()

0     TimRobbins,MorganFreeman,BobGunton,WilliamSadler
1          MarlonBrando,AlPacino,JamesCaan,DianeKeaton
2    ChristianBale,HeathLedger,AaronEckhart,Michael...
3       AlPacino,RobertDeNiro,RobertDuvall,DianeKeaton
4        HenryFonda,LeeJ.Cobb,MartinBalsam,JohnFiedler
Name: all_stars2, dtype: object

In [19]:
import ast

In [20]:
df_all_movies.all_stars2.apply(lambda x: x[0:-1].split(','))

0      [TimRobbins, MorganFreeman, BobGunton, William...
1        [MarlonBrando, AlPacino, JamesCaan, DianeKeato]
2      [ChristianBale, HeathLedger, AaronEckhart, Mic...
3      [AlPacino, RobertDeNiro, RobertDuvall, DianeKe...
4      [HenryFonda, LeeJ.Cobb, MartinBalsam, JohnFiedle]
5      [ElijahWood, ViggoMortensen, IanMcKellen, Orla...
6      [JohnTravolta, UmaThurman, SamuelL.Jackson, Br...
7      [LiamNeeson, RalphFiennes, BenKingsley, Caroli...
8      [LeonardoDiCaprio, JosephGordon-Levitt, Elliot...
9         [BradPitt, EdwardNorton, MeatLoaf, ZachGrenie]
10      [ElijahWood, IanMcKellen, OrlandoBloom, SeanBea]
11        [TomHanks, RobinWright, GarySinise, SallyFiel]
12     [ClintEastwood, EliWallach, LeeVanCleef, AldoG...
13     [ElijahWood, IanMcKellen, ViggoMortensen, Orla...
14     [LillyWachowski, KeanuReeves, LaurenceFishburn...
15     [RobertDeNiro, RayLiotta, JoePesci, LorraineBr...
16     [MarkHamill, HarrisonFord, CarrieFisher, Billy...
17     [JackNicholson, LouiseFl

In [21]:
df_all_movies['all_stars2'].head()

0     TimRobbins,MorganFreeman,BobGunton,WilliamSadler
1          MarlonBrando,AlPacino,JamesCaan,DianeKeaton
2    ChristianBale,HeathLedger,AaronEckhart,Michael...
3       AlPacino,RobertDeNiro,RobertDuvall,DianeKeaton
4        HenryFonda,LeeJ.Cobb,MartinBalsam,JohnFiedler
Name: all_stars2, dtype: object

In [22]:
df_all_movies['all_stars2'] = df_all_movies.all_stars2.apply(lambda x: x[0:-1].split(',')).apply(lambda x:['star_' + y for y in x if y])

Hello everyone. I'm work in a classification projecto of movies genere (using the imd-dataset: https://www.kaggle.com/harshitshankhdhar/imdb-dataset-of-top-1000-movies-and-tv-shows). 
I created a column with all stars that be in the movie from the columns Star1, Star2, Star3, Star4 and now I wanna use the get_dummy to have dummies colums about every star (ex. col_star_TimRobbins, col_star_MorganFreeman...).

But


In [23]:
df_all_movies['all_stars2'].head()

0    [star_TimRobbins, star_MorganFreeman, star_Bob...
1    [star_MarlonBrando, star_AlPacino, star_JamesC...
2    [star_ChristianBale, star_HeathLedger, star_Aa...
3    [star_AlPacino, star_RobertDeNiro, star_Robert...
4    [star_HenryFonda, star_LeeJ.Cobb, star_MartinB...
Name: all_stars2, dtype: object

In [24]:
(df_all_movies['all_stars2'].apply(pd.Series))

0                         1  \
0             star_TimRobbins        star_MorganFreeman   
1           star_MarlonBrando             star_AlPacino   
2          star_ChristianBale          star_HeathLedger   
3               star_AlPacino         star_RobertDeNiro   
4             star_HenryFonda            star_LeeJ.Cobb   
5             star_ElijahWood       star_ViggoMortensen   
6           star_JohnTravolta           star_UmaThurman   
7             star_LiamNeeson         star_RalphFiennes   
8       star_LeonardoDiCaprio  star_JosephGordon-Levitt   
9               star_BradPitt         star_EdwardNorton   
10            star_ElijahWood          star_IanMcKellen   
11              star_TomHanks          star_RobinWright   
12         star_ClintEastwood           star_EliWallach   
13            star_ElijahWood          star_IanMcKellen   
14        star_LillyWachowski          star_KeanuReeves   
15          star_RobertDeNiro            star_RayLiotta   
16            star_MarkHamill         star_HarrisonFord   
17         star_JackNicholson       star_LouiseFletcher   
18     star_Lin-ManuelMiranda          star_PhillipaSoo   
19           star_Kang-hoSong          star_LeeSun-kyun   
20                star_Suriya             star_Madhavan   
21    star_MatthewMcConaughey         star_AnneHathaway   
22             star_KátiaLund   star_AlexandreRodrigues   
23          star_DaveighChase     star_SuzannePleshette   
24              star_TomHanks            star_MattDamon   
25              star_TomHanks  star_MichaelClarkeDuncan   
26        star_RobertoBenigni     star_NicolettaBraschi   
27         star_MorganFreeman             star_BradPitt   
28           star_JodieFoster       star_AnthonyHopkins   
29            star_MarkHamill         star_HarrisonFord   
..                        ...                       ...   
956            star_BarryCook           star_Ming-NaWen   
957           star_JohnnyDepp       star_BenicioDelToro   
958        star_SusanneLothar           star_UlrichMühe   
959          star_RufusSewell     star_KieferSutherland   
960         star_RobertDeNiro           star_KevinBacon   
963          star_BruceWillis          star_JeremyIrons   
964           star_JohnnyDepp           star_GaryFarmer   
966             star_TomHanks           star_BillPaxton   
967  star_ZbigniewZamachowski           star_JulieDelpy   
968       star_MichaelDouglas         star_RobertDuvall   
971             star_MikiImai      star_ToshirôYanagiba   
975         star_BillyCrystal              star_MegRyan   
976           star_JohnMusker           star_JodiBenson   
979            star_MelGibson          star_DannyGlover   
980            star_EthanCoen             star_JohnGetz   
982            star_MelGibson          star_BruceSpence   
983          star_MichaelBeck           star_JamesRemar   
984            star_JimHenson              star_FrankOz   
985        star_ClintEastwood      star_PatrickMcGoohan   
986           star_JohnHubley             star_JohnHurt   
987            star_BradDavis         star_IreneMiracle   
989         star_ElliottGould      star_NinavanPallandt   
990           star_RodSteiger          star_JamesCoburn   
991        star_ClintEastwood         star_TellySavalas   
992           star_PhilHarris       star_SebastianCabot   
993        star_DavidHemmings      star_VanessaRedgrave   
994           star_JohnLennon        star_PaulMcCartney   
995        star_AudreyHepburn        star_GeorgePeppard   
997        star_BurtLancaster      star_MontgomeryClift   
999          star_RobertDonat     star_MadeleineCarroll   

                          2                         3  
0            star_BobGunton         star_WilliamSadle  
1            star_JamesCaan           star_DianeKeato  
2         star_AaronEckhart          star_MichaelCain  
3         star_RobertDuvall           star_DianeKeato  
4         star_MartinBalsam           star_JohnFiedle  
5          star_IanMcKellen      

In [25]:
type(df_all_movies)

pandas.core.frame.DataFrame

In [26]:
df_all_star = df_all_movies['all_stars2'].apply(pd.Series)

In [27]:
type(df_all_star)

pandas.core.frame.DataFrame

In [28]:
df_all_star.head()

0                   1                  2  \
0     star_TimRobbins  star_MorganFreeman     star_BobGunton   
1   star_MarlonBrando       star_AlPacino     star_JamesCaan   
2  star_ChristianBale    star_HeathLedger  star_AaronEckhart   
3       star_AlPacino   star_RobertDeNiro  star_RobertDuvall   
4     star_HenryFonda      star_LeeJ.Cobb  star_MartinBalsam   

                   3  
0  star_WilliamSadle  
1    star_DianeKeato  
2   star_MichaelCain  
3    star_DianeKeato  
4    star_JohnFiedle

In [29]:
df_all_star_one_collum = DataFrame(df_all_star[0]+','+ df_all_star[1]+',' +df_all_star[2]+','+ df_all_star[3])

NameError: name 'DataFrame' is not defined

In [ ]:
df_all_star_one_collum = DataFrame(pd.concat([df_all_star[0], df_all_star[1],df_all_star[2], df_all_star[3]]))

In [ ]:
type(df_all_star_one_collum)

In [ ]:
from pandas import Series, DataFrame

In [ ]:
type(df_all_star_one_collum)

In [ ]:
df_all_star_one_collum[0].str.value_counts()

We have a lot of 'Stars' with just one movie, so, lets removed this stars

In [ ]:
df_all_star_one_collum[df_all_star_one_collum.value_counts() <8]

In [ ]:
stars_to_remove = df_all_star_one_collum[df_all_star_one_collum.value_counts() <8]

In [ ]:
df = df[df.ids != "aball"]

In [ ]:
series_obj = Series

In [ ]:
stars_to_remove = df_all_star_one_collum.value_counts() <8

In [ ]:
stars_to_remove2 = df_all_star_one_collum[df_all_star_one_collum.value_counts() <8]

In [ ]:
stars_to_remove

In [ ]:
len(stars_to_remove)

In [ ]:
df_all_star_one_collum[df_all_star_one_collum == 'star_RobertDeNiro'].value_counts()

In [ ]:
stars_to_remove

In [ ]:
import re

In [ ]:
df_all_movies['all_star_3'] = df_all_movies['all_star_3'].str.replace('|'.join(map(re.escape, To_remove_lst)), '')

In [ ]:
df['name'] = df['name'].str.replace('|'.join(map(re.escape, To_remove_lst)), '')

In [ ]:
df_all_movies.set_index('Series_Title', drop=False, append=True)

In [ ]:
print(df_all_movies['all_stars2'][3])

In [ ]:
dummie2 = pd.get_dummies(df_all_movies['all_stars2'].apply(pd.Series))

In [ ]:
dummie2

In [ ]:
#dummie = pd.get_dummies(df_all_movies['all_stars2'].apply(pd.Series)).sum(level=0)

In [ ]:
df_all_movies = pd.concat([df_all_movies, dummie2], axis = 1)

In [ ]:
df_all_movies['2_star_BruceWillis']

In [ ]:
with pd.option_context('display.max_columns',None):
    display(df_all_movies.head())

In [ ]:
df_all_movies.head()

In [ ]:
df_all_movies.drop()

so we created mora than 3000 variables, its not sound good

In [ ]:
dummie.head()

In [ ]:
#df_all_movies_2 = pd.get_dummies(df_all_movies['all_stars2'].apply(pd.Series)).sum(level=0)


In [ ]:
df_all_movies_2.head()

In [ ]:
df_all_movies_2

In [ ]:
df_all_movies_2 = df_all_movies.copy()

In [ ]:
df_all_movies_2

In [ ]:
from IPython.display import display

In [ ]:
with pd.option_context('display.max_columns',None):
    display(df_all_movies.head())

In [ ]:
df_all_movies.head()

### EDA

In [ ]:
continuous = [f for f in df_all_movies.columns if df_all_movies.dtypes[f] != 'object']
continuous